File: php_with_pecl_http/rp_del_contact.php
#!/usr/bin/php
<?php
// Program: rp_del_contact.php
// Usage: ./rp_del_contact.php
// Date: 2020 10 07
// Purpose: php language sample client program for RackPing Monitoring API 2.0
// Version: 1.0
// Copyright: RackPing USA 2020
// Env: PHP5 with pecl_http or newer
// Returns: exit status is non-zero on failure
// Note: first do: source ../set.sh

   error_reporting(E_ALL);

   $url       = getenv('RP_SCHEME') . getenv('RP_DOMAIN') . getenv('RP_BASE_URL');
   $api_key   = getenv('RP_API_KEY');
   $user      = getenv('RP_USER');
   $password  = getenv('RP_PASSWORD');
   $timeout   = getenv('RP_TIMEOUT');
   $redirects = getenv('RP_REDIRECTS');
   $useragent = getenv('RP_USERAGENT');
   $debug     = getenv('RP_DEBUG');

   if (empty($api_key)) {
      echo "error: do source ../set.sh to set the envariables.\n";
      exit(1);
   }

   $id = $argv[1];

   if (empty($id)) {
      echo "usage: ./$argv[0] contactid\n";
      exit(1);
   }

   fwrite(STDERR, "Delete one contact" . PHP_EOL);

   $http_req = new HttpRequest($url . '/contacts/' . $id, HttpRequest::METH_DELETE);

   $headers = array(
      'App-key'       => $api_key,
      'Authorization' => "Basic " . base64_encode("$user:$password"),
      'Accept'        => 'application/json',
      'Accept-Charset' => 'utf-8'
   );

   $http_req->setHeaders($headers);

   $http_req->setOptions(array('timeout'   => $timeout,
                               'useragent' => $useragent,
                               'redirect'  => $redirects
   ));

   $ret = 0;

   try {
       $http_req->send();
       if ($http_req->getResponseCode() == 200) {
           echo $http_req->getResponseBody();
       }
       else {
           if ($debug) {
              echo $http_req->getRawResponseMessage();
           }
           echo "HTTP response code: " . $http_req->getResponseCode() . "\n";
           exit(1);
       }
   } catch (HttpException $ex) {
       echo $ex;
       exit(1);
   }

   exit($ret);
?>

File: golang/rp_del_contact.go
//usr/bin/go run $0 $@; exit $?

// Program: rp_del_contact.go
// Usage: go run rp_del_contact.go id
// Purpose: Go language sample client program for RackPing Monitoring API 2.0
// Version: 1.0
// Copyright: RackPing USA 2020
// Env: Go 1.13.14 or higher
// Returns: exit status is non-zero on failure
// Notes: first set envariables with: source ../set.sh

package main

import (
//      "bytes"
//      "encoding/json"
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
        "net/http/httputil"
	"os"
        "strconv"
	"time"
)

func debug(data []byte, err error) {
    if err == nil {
       fmt.Printf("%s\n\n", data)
    } else {
       log.Fatalf("%s\n\n", err)
    }
}

// global variables for communicating with callback redirect function redirectPolicyFunc()
var g_username   string = ""
var g_password   string = ""
var g_api_key    string = ""
var g_user_agent string = ""

func redirectPolicyFunc(req *http.Request, via []*http.Request) error {

   return nil
}

func main() {
   g_username   = os.Getenv("RP_USER")
   g_password   = os.Getenv("RP_PASSWORD")
   g_api_key    = os.Getenv("RP_API_KEY")
   g_user_agent = os.Getenv("RP_USERAGENT")

   if g_username == "" {
      fmt.Printf("%s\n", "error: do 'source ../set.sh' first.")
      os.Exit(1)
   }

   var domain   string = os.Getenv("RP_DOMAIN")
   var base_url string = os.Getenv("RP_BASE_URL")
   var scheme   string = os.Getenv("RP_SCHEME")

   var Debug bool = (os.Getenv("RP_DEBUG") == "1")

   // see https://golang.org/pkg/time/ for how time casts work
   timeout, err := strconv.Atoi(os.Getenv("RP_TIMEOUT"))
   if err != nil {
      log.Fatal(err)
   }

   var url string = scheme + domain + base_url

   client := &http.Client{
      Timeout: time.Duration(timeout) * time.Second,
      CheckRedirect: redirectPolicyFunc,
   }

   {
      if (len(os.Args) < 2) {
         fmt.Fprintf(os.Stderr, "usage: %s id\n", os.Args[0])
         os.Exit(0)
      }

      id := os.Args[1]

      fmt.Printf("%s\n", "Delete one contact")

      req, err := http.NewRequest("DELETE", url + "/contacts/"+id, nil)
      req.SetBasicAuth(g_username, g_password)
      req.Header.Set("app-key", g_api_key)
      req.Header.Set("Accept", "application/json")
      req.Header.Set("Accept-Charset", "utf-8")
      req.Header.Set("User-Agent", g_user_agent)

      if Debug { debug(httputil.DumpRequestOut(req, true)) }

      res, err := client.Do(req)

      if err != nil {
         log.Fatal(err)
         debug(httputil.DumpResponse(res, true))
         os.Exit(1)
      } else {
         defer res.Body.Close()
         if Debug { debug(httputil.DumpResponse(res, true)) }
         body, err := ioutil.ReadAll(res.Body)
         if err == nil && res.StatusCode == 200 {
            fmt.Printf("%s\n", body)
         } else {
            println("HTTP status code is", res.StatusCode)
            os.Exit(1)
         }
      }
   }

   os.Exit(0)
}

